m6-advanced-multithreading/t7-forkjoin/src/main/java/pl/mentoring/forkjoin/ForkBlur.java

```diff
@@ -0,0 +1,67 @@
+package pl.mentoring.forkjoin;
+
+import java.util.concurrent.RecursiveAction;
+
+/**
+ * ForkBlur implements a simple horizontal image blur. It averages pixels in the
+ * source array and writes them to a destination array. The sThreshold value
+ * determines whether the blurring will be performed directly or split into two
+ * tasks.
+ * <p>
+ * This is not the recommended way to blur images; it is only intended to
+ * illustrate the use of the Fork/Join framework.
+ */
+public class ForkBlur extends RecursiveAction {
+
+    private int[] mSource;
+    private int mStart;
+    private int mLength;
+    private int[] mDestination;
+    private int mBlurWidth = 15; // Processing window size, should be odd.
+
+    public ForkBlur(int[] src, int start, int length, int[] dst) {
+        mSource = src;
+        mStart = start;
+        mLength = length;
+        mDestination = dst;
+    }
+
+    // Average pixels from source, write results into destination.
+    protected void computeDirectly() {
+        int sidePixels = (mBlurWidth - 1) / 2;
+        for (int index = mStart; index < mStart + mLength; index++) {
+            // Calculate average.
+            float rt = 0, gt = 0, bt = 0;
+            for (int mi = -sidePixels; mi <= sidePixels; mi++) {
+                int mindex = Math.min(Math.max(mi + index, 0), mSource.length - 1);
+                int pixel = mSource[mindex];
+                rt += (float) ((pixel & 0x00ff0000) >> 16) / mBlurWidth;
+                gt += (float) ((pixel & 0x0000ff00) >> 8) / mBlurWidth;
+                bt += (float) ((pixel & 0x000000ff) >> 0) / mBlurWidth;
```

Comment:
What does (pixel & 0x000000ff) >> 0 do?

Reply:
My guess - makes formulas uniform and easier to read.
It's code from tutorial, I used it as is: https://docs.oracle.com/javase/tutorial/essential/concurrency/forkjoin.html